presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -160,37 +165,51 @@ public Page getOutput()
             checkState(process != null, "process is null");
             createTask();
             checkState(task != null, "task is null");
-            taskStatusFuture = task.start();
+            TaskInfo taskInfo = task.start();
```

Comment:
Should we ideally move this to `noMoreSplits`, or someplace we know is going to called once, and after all splits are added to operator

Reply:
We can do that, but probably in a separate PR.

Reply:
Actually, It should be feasible to move this whole block in `noMoreSplits()`  
```Java
if (process == null) {
    createProcess();
    checkState(process != null, "process is null");
    createTask();
    checkState(task != null, "task is null");
    TaskInfo taskInfo = task.start();
    if (processTaskInfo(taskInfo)) {
       return null;
    }
}
```